woodwork/type_sys/type_system.py

```diff
@@ -298,7 +298,7 @@ def infer_logical_type(self, series):
             series = series.head(INFERENCE_SAMPLE_SIZE)
         else:
             if _is_dask_series(series):
-                series = series.head(INFERENCE_SAMPLE_SIZE)
+                series = series.compute().head(INFERENCE_SAMPLE_SIZE)
```

Comment:
I'm sure you know this already, but this `.compute()` call could be quite costly in terms of performance. Not sure how this would compare to using `series.head(INFERENCE_SAMPLE_SIZE, npartitions=-1)` in terms of performance, but I think you would get the same result in both cases.